src/kOS.Safe/Communication/Connection.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+
+namespace kOS.Safe.Communication
+{
+    [kOS.Safe.Utilities.KOSNomenclature("Connection")]
+    public abstract class Connection<S> : Structure where S : SharedObjects
```

Comment:
Yeah, `ProcessorConnection` and `VesselConnection` both need to access fields `kOS.SharedObjects`.

I just had another look at the `Connection` class. The `shared` field is not used in there, only in the subclasses, so I've moved it there. I think the `<S>` parameter was a remnant of some older code that I must have had refactored at some point. Anyway, I have removed it now.